src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt

```diff
@@ -209,3 +214,11 @@ data class ScheduleDomainEvent(
 data class ScheduleAdditionalInformation(
   val activityScheduleId: Long,
 )
+
+fun ActivitySchedule.toCreatePayPerSession(): CreateActivityRequest.PayPerSession =
+  CreateActivityRequest.PayPerSession.values()
+    .first { it.value == this.activity.payPerSession.value }
+
+fun ActivitySchedule.toUpdatePayPerSession(): UpdateActivityRequest.PayPerSession =
+  UpdateActivityRequest.PayPerSession.values()
+    .first { it.value == this.activity.payPerSession.value }
```

Comment:
Despite reusing the same enum for PayPerSession the Open API generator creates multiple copies of it, so we need to map it the Activities enum for each copy.